### PR TITLE
proper YAML syntax for rewriting ansible tasks

### PR DESCRIPTION
**Pull request by arbabnazar**
@edx/devops kindly review. Thanks

### CODE REVIEW

File: playbooks/roles/minos/tasks/main.yml [hunk 1]
```diff
@@ -35,61 +35,65 @@
...
-    mode=750
-  when:
-    - MINOS_GIT_IDENTITY is not defined
+# Optional auth for git
+- name: Create ssh script for git (not authenticated)
+  template:
+    src: "tmp/git-identity.sh.j2"
+    dest: "{{ minos_git_ssh }}"
+    mode: "0750"
+  when: MINOS_GIT_IDENTITY is not defined
```
Comment (jibsheet) [inline playbooks/roles/minos/tasks/main.yml]: Not part of your cleanup, but these tasks appear identical (and the template already checks for MINOS_GIT_IDENTITY) maybe collapse these without the whens?

the not authenticated and authenticated cases for clarity

**Comment by jibsheet** [comment]
👍 with my note about squishing two tasks

File: playbooks/roles/minos/tasks/main.yml (hunk 1 above)
Comment (arbabnazar) [inline playbooks/roles/minos/tasks/main.yml]: @jibsheet when statement is there, sorry If I misunderstood. Can you please explain a bit, may be I got it wrong. Thanks

Comment (jibsheet) [inline playbooks/roles/minos/tasks/main.yml]: Sorry, I meant the two tasks look identical.  One run when defined, one when not defined.  But either way, they do the same thing.  I can't tell if I'm misreading the code, or if they're identical and we should have 1 task without the when.  Does that make sense?

Comment (arbabnazar) [inline playbooks/roles/minos/tasks/main.yml]: Thanks @jibsheet, we can ask from @e0d and @feanil, they can tell us.

Comment (arbabnazar) [inline playbooks/roles/minos/tasks/main.yml]: @e0d @feanil can you please guide here. Thanks

**Comment by arbabnazar** [comment]
@jibsheet I have addressed your comment and removed the identical tasks and when, please review. Thanks

CC: @edx/devops

**Comment by jibsheet** [comment]
👍

**Pull request merged**